src/basic/pidfd-util.c

```diff
@@ -170,6 +170,60 @@ int pidfd_verify_pid(int pidfd, pid_t pid) {
         return current_pid != pid ? -ESRCH : 0;
 }
 
+int pidfd_get_ppid(int fd, pid_t *ret) {
+        struct pidfd_info info = { .mask = PIDFD_INFO_PID };
```

Comment:
doesn't matter, but i'd move this down, immediately before the pidfd_get_info() where you need it.

Reply:
I prefer the current style better. The function is small and the var is used all across it after all.